### ISSUE TITLE
Strange crosstalk(?} issue with deadpool redis

### ISSUE THREAD

**Issue by henrikrudstrom** [bikeshedder/deadpool#102]
We have a service in development using deadpool_redis. It seems to run fine and do its thing, but after a couple of days of running it starts misbehaving: 
When the code perfoms a `GET` on a key (no matter if it exists or not), it returns the value `PONG` (we do one `PING` at startup), its not a transient issue, but it persists until we restart the service. 

```rust
let mut conn = self.pool.get().await.map_err(|e| Error::Pool(e))?;
        let query = format!("frontpage-ids/{}", frontpage_id);
        debug!(self.logger, "redis query: {}", query);
        let data: String = cmd("GET")  // <- this one returs PONG
            .arg(&query)
            .query_async(&mut conn)
            .await
            .map_err(|e| Error::Redis(format!("GET {}", query), e))?;

        deserialize_frontpage_teasers(data, max_items)
```

we initalize the pool with this code:
```rust
        let mut cfg = Config::default();
        let rurl = format!("redis://{}", rcfg.redis.url);
        cfg.url = Some(rurl.to_string());
        let pool = cfg
            .create_pool()
            .map_err(|e| Error::Redis("connect".to_string(), e))?;
        Ok(Self {
            pool: pool,
            logger: logger,
        })
```

I find this pretty mindboggling and its frustrating that the error takes many days to manifest, making debugging a *very* slow process. Any clues would be greatly appreciated...

using deadpool_redis version 0.6

**Comment by bikeshedder** [comment]
This was reported as part of #97 and is caused by a known issue of the `redis` crate: 
- mitsuhiko/redis-rs#489
- mitsuhiko/redis-rs#325

Aborted futures can leave the redis connection in a unusable state.

I released patched versions of `deadpool-redis` for `0.6`, `0.7` and `0.8`. Those versions send a payload with the `PING` command and ensure that the `PONG` response return the same number.

Upgrading to [`deadpool-redis 0.6.2`](https://crates.io/crates/deadpool-redis/0.6.2) should fix that issue: https://github.com/bikeshedder/deadpool/commit/7136652b1660ee7632c4a5e453d0dec0bb083b97 https://github.com/bikeshedder/deadpool/commit/36ffd106b71781b9ffa4081c14ef30f2b7b9474d

Be warned though - if you reuse connections without using the pool you either need to wrap all redis operations inside a `tokio::spawn(async move { ... })` or ensure somehow else that a future executing a redis command is always polled to completion before issuing the next command.

**Comment by henrikrudstrom** [comment]
Great! Thanks for the fast response.
